plugins/redfish/fu-redfish-client.c

```diff
@@ -76,34 +101,62 @@ fu_redfish_client_coldplug_member (FuRedfishClient *self,
 				   JsonObject *member,
 				   GError **error)
 {
-	g_autoptr(FuDevice) dev = fu_device_new ();
+	g_autoptr(FuDevice) dev = NULL;
+	const gchar *guid = NULL;
+	g_autofree gchar *id = NULL;
+
+	if (json_object_has_member (member, "SoftwareId")) {
+		guid = json_object_get_string_member (member, "SoftwareId");
+	} else if (json_object_has_member (member, "Oem")) {
+		JsonObject *oem = json_object_get_object_member (member, "Oem");
+		if (json_object_has_member (oem, "Hpe")) {
+			JsonObject *hpe = json_object_get_object_member (oem, "Hpe");
+			if (json_object_has_member (hpe, "DeviceClass"))
+				guid = json_object_get_string_member (hpe, "DeviceClass");
+		}
+	}
+
+	/* Skip the devices without guid */
+	if (guid == NULL)
+		return TRUE;
 
+	dev = fu_device_new ();
+
+	id = g_strdup_printf ("Redfish-Inventory-%s",
+			      json_object_get_string_member (member, "Id"));
+	fu_device_set_id (dev, id);
+
+	fu_device_add_guid (dev, guid);
 	fu_device_set_name (dev, json_object_get_string_member (member, "Name"));
 	fu_device_set_summary (dev, "Redfish device");
 	fu_device_set_version (dev, json_object_get_string_member (member, "Version"));
 	if (json_object_has_member (member, "LowestSupportedVersion"))
 		fu_device_set_version_lowest (dev, json_object_get_string_member (member, "LowestSupportedVersion"));
 	if (json_object_has_member (member, "Description"))
 		fu_device_set_description (dev, json_object_get_string_member (member, "Description"));
-	if (json_object_get_boolean_member (member, "Updateable"))
+	if (json_object_has_member (member, "Updateable")) {
+		if (json_object_get_boolean_member (member, "Updateable"))
+			fu_device_add_flag (dev, FWUPD_DEVICE_FLAG_UPDATABLE);
```

Comment:
Something we have done recently in another plugin was to set the update error state when the device isn't updatable for the reason it's not updatable.  For example if the reason that the json didn't have an Updateable member was because of a BIOS option for example that would be a good reason to advertise here in this situation.

Reply:
I've contacted the HPE engineer, and he said the Updateable field is not implemented yet, so I really don't know if they are updatable or not. If this could be a problem, I'll remove the bit if Updateable is absent.